### ISSUE TITLE
setup vue-styleguidist with vue-cli

### ISSUE THREAD

**Issue by nicokoenig** [vue-styleguidist/vue-styleguidist#150]
Hey y'all,

I tried to setup vue-styleguidist with a vue-cli based project following the vue-styleguidist docs.

### Setup
* vue-cli version: 3.0.0-rc.3
* vue-styleguidist version: 1.7.10

I ran vue-cli with 
> vue create styleguide

and used the presets (babel + eslint).

### Error
When running `npm run styleguide` I get the following error:

```
./src/components/HelloWorld.vue?vue&type=template&id=656039f0&scoped=true 2:0
Module parse failed: Unexpected token (2:0)
You may need an appropriate loader to handle this file type.
|
> <div class="hello">
|   <h1>{{ msg }}</h1>
|   <p>

Learn how to add webpack loaders to your style guide:
https://github.com/vue-styleguidist/vue-styleguidist/blob/master/docs/Webpack.md

```

### Possible solutions

I read through the following docs 

* https://github.com/vue-styleguidist/vue-styleguidist/blob/master/docs/Webpack.md
* https://github.com/vue-styleguidist/vue-styleguidist/blob/master/docs/GettingStarted.md

but could not find a solution.

I also tried the create an styleguide.config.js / webpack.config.js that points to the vue-cli webpack config, but still I get the same error.

```
module.exports = {
    webpackConfig: require('node_modules/@vue/cli-service/webpack.config.js')
 };

```
Am I missing something here?

Cheers,
Nico

**Comment by nicokoenig** [comment]
Your fix works perfectly ❤️

**Comment by rafegoldberg** [comment]
Using Vue's built-in `~` and `@` alias in my component SCSS was causing vue-styleguidist` to throw a similar error to the OP:

```
./src/components/UI/Panel/index.vue?vue&type=style&index=0&id=54b531fc&lang=scss&scoped=true 14:0
Module parse failed: Unexpected character '@' (14:0)
You may need an appropriate loader to handle this file type.
|
|
> @import "~@/styles/config/scale";
| .UiPanel {
|
error Command failed with exit code 1.
```

I figured this stemmed from some internal mis-handling of my project's Webpack config. So after some poking around, I tried checking out f08398b. This seemed to solve my Webpack loader issues (wahoo!), but now when I run the `build` task I get a different error... I can't make much sense of it, but maybe @rafaesc can:

```
FAIL  Failed to compile

 multi ./node_modules/vue-styleguidist/lib/index ./node_modules/react-dev-utils/webpackHotDevClient.js
 Module not found: Can't resolve '/Users/z00221y/.bitnami/stackman/machines/xampp/volumes/root/htdocs/clients/ws3/wp-content/themes/ws-ui/node_modules/vue-styleguidist/lib/index' in '/Users/z00221y/.bitnami/stackman/machines/xampp/volumes/root/htdocs/clients/ws3/wp-content/themes/ws-ui'
```

**Comment by rafaesc** [comment]
@rafegoldberg do you use vue-cli? Does the problem only appear when you try to build or also with you want to run server?

**Comment by rafegoldberg** [comment]
@rafaesc yep, I'm using vue-cli. My setup looks as follows:

- `@vue/cli-service@3.0.0-rc.3`
- `vue@2.5.16`
- `vue-styleguidist@1.7.10` 

To your question re: `serve` versus `build, the error logs I posted were from the former task. (I haven't tried running the full build yet...)

**Comment by rafegoldberg** [comment]
@rafaesc running the Styleguidist `build` on f08398b returns a similar error to the `serve` task:

```
FAIL  Failed to compile

 multi ./node_modules/vue-styleguidist/lib/index
 Module not found: Can't resolve '/Users/z00221y/.bitnami/stackman/machines/xampp/volumes/root/htdocs/clients/ws3/wp-content/themes/ws-ui/node_modules/vue-styleguidist/lib/index' in '/Users/z00221y/.bitnami/stackman/machines/xampp/volumes/root/htdocs/clients/ws3/wp-content/themes/ws-ui'
```

**Comment by rafegoldberg** [comment]
(Also, if somewhat obviously, when I `ls ./node_modules/vue-styleguidist/` the `lib/` directory which the task is looking for doesn't exist at all!)

**Comment by rafegoldberg** [comment]
Also, I just noticed that right after I start either of the `vue-styleguidist` tasks, it throws a deprecation warning:

```
(node:58289) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
```

Not sure if that's related tho...

**Comment by rafaesc** [comment]
@rafegoldberg yes, its because vue-styleguidist's components haven't been compiled yet in your environment.

But I will release a new version, to avoid to install vue-styleguidist dependency through github

**Comment by rafaesc** [comment]
@nicokoenig @rafegoldberg 
Fixed on 1.7.11 version

**Comment by rafegoldberg** [comment]
@rafaesc really appreciate the help here, and the (stunningly) quick turnaround on the `1.7.11` release. Thanks for all the work on vue-styleguidist my man!
